In [1]:
#Step1: Gathering Data
import numpy as np
import pandas as pd

In [2]:
data=pd.read_csv('/content/insurance.csv')


In [3]:
data.head()

age     sex   bmi  children smoker     region  expenses
0   19  female  27.9         0    yes  southwest  16884.92
1   18    male  33.8         1     no  southeast   1725.55
2   28    male  33.0         3     no  southeast   4449.46
3   33    male  22.7         0     no  northwest  21984.47
4   32    male  28.9         0     no  northwest   3866.86

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   expenses  1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [5]:
#step2: Data pre-processing where we will be using one hot encoding method
data['sex']=data['sex'].apply({'male':1,'female':0}.get)

In [6]:
data.head()

age  sex   bmi  children smoker     region  expenses
0   19    0  27.9         0    yes  southwest  16884.92
1   18    1  33.8         1     no  southeast   1725.55
2   28    1  33.0         3     no  southeast   4449.46
3   33    1  22.7         0     no  northwest  21984.47
4   32    1  28.9         0     no  northwest   3866.86

In [7]:
data['smoker']=data['smoker'].apply({'yes':1,'no':0}.get)

In [8]:
data.head()

age  sex   bmi  children  smoker     region  expenses
0   19    0  27.9         0       1  southwest  16884.92
1   18    1  33.8         1       0  southeast   1725.55
2   28    1  33.0         3       0  southeast   4449.46
3   33    1  22.7         0       0  northwest  21984.47
4   32    1  28.9         0       0  northwest   3866.86

In [9]:
data['region']=data['region'].apply({'southwest':0,'southeast':1,'northwest':2,'northeast':3}.get)

In [10]:
data.head()

age  sex   bmi  children  smoker  region  expenses
0   19    0  27.9         0       1       0  16884.92
1   18    1  33.8         1       0       1   1725.55
2   28    1  33.0         3       0       1   4449.46
3   33    1  22.7         0       0       2  21984.47
4   32    1  28.9         0       0       2   3866.86

In [11]:
#step3: Dividing data into dependent and independent data
x=data[['age','sex','bmi','children','smoker','region']]#dependent variables

In [12]:
y=data[['expenses']]#independent variables

In [13]:
#Splitting the data into training and testing data
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2)

In [14]:
from sklearn.linear_model import LinearRegression
linear_model=LinearRegression()

In [15]:
#training the model using fit()
linear_model.fit(x_train,y_train)

LinearRegression()

In [32]:
y_pred=linear_model.predict(x_test)


In [33]:
print(y_pred)#y_pred is the amount of health insurance as calculated by the ML model

[[ 7611.01050182]
 [ 7302.67222712]
 [ 7807.48031008]
 [13001.58280524]
 [14813.61577151]
 [15387.69405451]
 [ 1881.40057701]
 [39067.6013173 ]
 [ 8157.86963863]
 [ 2648.62434349]
 [  386.80254773]
 [ 7268.57722723]
 [ 2811.76156259]
 [ 2293.92599662]
 [38877.76806947]
 [11219.81692097]
 [ 8420.79578309]
 [ 2336.42218847]
 [ 6107.20768611]
 [-2208.88336581]
 [10421.79229561]
 [10020.39768596]
 [ 1464.06311333]
 [ 2255.30051885]
 [  248.72433522]
 [ 1902.81032908]
 [ 7145.56586169]
 [ 7946.65655657]
 [ 9948.84113258]
 [25809.08232909]
 [ 8256.87340969]
 [ 7430.66360159]
 [33736.61415032]
 [13132.83896522]
 [10339.04482563]
 [ 6569.45343103]
 [13520.40828106]
 [ 6961.07955951]
 [ 8461.8876682 ]
 [ 9628.39996679]
 [33617.40134217]
 [ 5514.36528914]
 [ 7424.20735606]
 [ 7452.58381459]
 [28093.42400004]
 [14219.02161632]
 [  131.34752611]
 [29823.42415046]
 [18410.36735616]
 [ 9181.22106885]
 [11620.22161576]
 [ 2163.18608126]
 [ 8722.45552443]
 [11670.26003706]
 [12374.41185967]
 [10911.78

In [34]:
#checking accuracy of ML model
accuracy=linear_model.score(x,y)


In [35]:
print('The accuracy of the ML model using linear Regression is',accuracy)

The accuracy of the ML model using linear Regression is 0.7505978067419891


In [41]:
#predicting the health insurance of a new customer
data_new={'age':32,'sex':0,'bmi':22,'children':1,'smoker':0,'region':3}

In [42]:
#converting dictinary into data frame
index=[1]#serial number
my_data=pd.DataFrame(data_new,index)


In [43]:
my_data

age  sex  bmi  children  smoker  region
1   32    0   22         1       0       3

In [44]:
#predicting the amount of health insurance the new customer is going to get
y_pred=linear_model.predict(my_data)

In [45]:
print('the amount of health insurance the customer is going to get',y_pred)

the amount of health insurance the customer is going to get [[4014.27100632]]


**Why is this $4014 is low for this lady?**
ans= because she is actually healthy and has only 1 kid and she is also non-smoker